crates/solver/src/settlement_access_list.rs

```diff
@@ -75,21 +138,26 @@ pub struct TenderlyApi {
 
 impl TenderlyApi {
     #[allow(dead_code)]
-    pub fn new(url: Url, api_key: &str, network_id: String) -> Self {
-        Self {
-            url,
-            client: Client::new(),
+    pub fn new(
+        url: impl IntoUrl,
+        client: Client,
+        api_key: &str,
+        network_id: String,
+    ) -> Result<Self> {
+        Ok(Self {
+            url: url.into_url()?,
+            client,
             header: {
                 let mut header = HeaderMap::new();
                 header.insert("x-access-key", HeaderValue::from_str(api_key).unwrap());
```

Comment:
Since the function now returns a `Result` I think it would be nice to also have `HeaderValue::from_str(api_key)?`.

Reply:
fixed, thanks